src/Support/Container.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Support;
+
+use Pest\Exceptions\ContainerTypeCannotBeDetermined;
+use Pest\Exceptions\ContainerValueCannotBeResolved;
+use Psr\Container\ContainerInterface;
+use ReflectionClass;
+use ReflectionParameter;
+
+/**
+ * @internal
+ */
+final class Container implements ContainerInterface
+{
+    /**
+     * @var array<string, mixed>
+     */
+    private $instances = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function has($id): bool
+    {
+        return array_key_exists($id, $this->instances);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return mixed
+     */
+    public function get($id)
+    {
+        if ($this->has($id)) {
+            return $this->instances[$id];
+        }
+
+        $this->instances[$id] = $this->build($id);
```

Comment:
No need for the `build` thing. If the instance do not exist let's throw a `ShouldNotHappen` in this cases.

Reply:
I think we need the `build` method because we need it for the instance creation of the Plugins. Otherwise we need to push the container to the plugins via constructor.

Reply:
Is pushing the container to plugins through the constructor an issue? 🤔 I feel like accessing `$this->container->get()` in a plugin class seems fine. Or have I misunderstood this? 🙂

Reply:
Injecting a container in classes is bad practice. Classes should declare their dependencies ( in your case the console output and the test suite ), and the container should resolve the dependencies while creating a new instance of that class.

Reply:
Thanks for the explanation 👍